schemachange: unskip TestDropWhileBackfill

spaskob opened the pull request:
Disabling the GC job was preventing this test from completing.
Tested with `test stress`: 1000 successful runs.

Fixes #44944.

Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49870)
<!-- Reviewable:end -->

thoszhang reviewed (approved):
LGTM, but you have some extra .tmp files in this commit.

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49870#-:-M9-7Op764an9-Yt1e_B:bsbf84h)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

Commit:
schemachange: unskip TestDropWhileBackfill

Disabling the GC job was preventing this test from completing.
Tested with `test stress`: 1000 successful runs.

Fixes #44944.

Release note: none.

spaskob commented:
bors r+
